db/migrations/1726615439_init.up.sql

```diff
@@ -1,58 +1,68 @@
+  display_name varchar(50) NOT NULL UNIQUE,
   email varchar(255) NOT NULL UNIQUE,
-  join_date timestamp NOT NULL DEFAULT now()
+  join_date timestamp NOT NULL DEFAULT now(),
+  updated_at timestamp NULL 
 );
+
 CREATE TABLE IF NOT EXISTS recipes (
   id bigserial PRIMARY KEY,
   author_id bigint NOT NULL CONSTRAINT fk_recipe_author REFERENCES users(id),
   forked_from bigint CONSTRAINT fk_recipe_fork REFERENCES recipes(id),
   slug varchar(75) NOT NULL UNIQUE,
-  description text,
+  private boolean NOT NULL,
   initial_publish_date timestamp NOT NULL DEFAULT now()
 );
-CREATE TABLE IF NOT EXISTS recipe_comments (
-  id bigserial PRIMARY KEY,
-  recipe_id bigint NOT NULL CONSTRAINT fk_recipe_comment REFERENCES recipes(id),
-  author_id bigint NOT NULL CONSTRAINT fk_recipe_comment_author REFERENCES users(id),
-  content text NOT NULL,
-  post_date timestamp NOT NULL DEFAULT now()
-);
+
 CREATE TABLE IF NOT EXISTS recipe_revisions (
   id bigserial PRIMARY KEY,
   recipe_id bigint NOT NULL CONSTRAINT fk_recipe_revisions REFERENCES recipes(id),
-  parent_id bigint NOT NULL CONSTRAINT fk_recipe_revision_parent REFERENCES recipe_revisions(id),
-  child_id bigint CONSTRAINT fk_recipe_revision_child REFERENCES recipe_revisions(id),
-  -- Free form content, maybe like an "about" section. Maybe this should be like explaining the changes made
-  description text,
+  parent_id bigint NULL CONSTRAINT fk_recipe_revision_parent REFERENCES recipe_revisions(id),
+  -- About section for the recipe.
+  recipe_description text,
+  --change description should be a note describing how description changed
+  change_comment text,
+  title text NOT NULL,
   publish_date timestamp NOT NULL DEFAULT now()
 );
+
+ALTER TABLE IF EXISTS recipes
+ADD featured_revision bigint NULL CONSTRAINT fk_recipe_revision_id REFERENCES recipe_revisions(id);
+
+   
 CREATE TABLE IF NOT EXISTS linked_recipes (
   from_recipe_id bigint NOT NULL CONSTRAINT fk_linked_recipes_from REFERENCES recipes(id),
   to_recipe_id bigint NOT NULL CONSTRAINT fk_linked_recipes_to REFERENCES recipes(id),
   CONSTRAINT linked_recipe_pk PRIMARY KEY(from_recipe_id, to_recipe_id)
 );
 CREATE TABLE IF NOT EXISTS tags (
-  name varchar(255) PRIMARY KEY,
-  description text
+  id bigserial PRIMARY KEY,
+  name varchar(255) NOT NULL UNIQUE,
+  description text,
+  --True if by user , False if internal keyword
+  user_generated boolean NOT NULL
 );
 CREATE TABLE IF NOT EXISTS measurement_units (
-  name varchar(255) PRIMARY KEY,
+  id bigserial PRIMARY KEY,
+  name varchar(255) NOT NULL UNIQUE,
   description text
 );
 CREATE TABLE IF NOT EXISTS measurement_units_tags (
-  measurement varchar(255) NOT NULL CONSTRAINT fk_measurement_tags_measurement REFERENCES measurement_units(name),
-  tag varchar(255) NOT NULL CONSTRAINT fk_measurement_tags_tag REFERENCES tags(name)
+  id bigserial PRIMARY KEY,
+  measurement bigint NOT NULL UNIQUE CONSTRAINT fk_measurement_tags_measurement REFERENCES measurement_units(id),
```

Comment:
I don't think we want a `UNIQUE` constraint here, since a measurement can have many tags. If anything we might want a composite constraint on both the measurement and tag, but we can always do that later.